test/es-module/test-esm-loader-modulemap.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+// Flags: --expose-internals
+
+// This test ensures that the type checking of ModuleMap throws
+// errors appropriately
+
+const common = require('../common');
+
+const { URL } = require('url');
+const Loader = require('internal/loader/Loader');
+const ModuleMap = require('internal/loader/ModuleMap');
+const ModuleJob = require('internal/loader/ModuleJob');
+const { createDynamicModule } = require('internal/loader/ModuleWrap');
+
+const stubModuleUrl = new URL('file://tmp/test');
+const stubModule = createDynamicModule(['default'], stubModuleUrl);
+const loader = new Loader();
+const moduleMap = new ModuleMap();
+const moduleJob = new ModuleJob(loader, stubModule.module);
```

Comment:
CI failed across all platforms:

```
not ok 2040 es-module/test-esm-loader-modulemap
  ---
  duration_ms: 0.72
  severity: fail
  stack: |-
    internal/loader/ModuleJob.js:18
        this.modulePromise = this.moduleProvider(url);
                                  ^
    
    TypeError: this.moduleProvider is not a function
        at new ModuleJob (internal/loader/ModuleJob.js:18:31)
        at Object.<anonymous> (/data/iojs/build/workspace/node-test-commit-linuxone/nodes/rhel72-s390x/test/es-module/test-esm-loader-modulemap.js:19:19)
        at Module._compile (module.js:616:30)
        at Object.Module._extensions..js (module.js:627:10)
        at Module.load (module.js:535:32)
        at tryModuleLoad (module.js:478:12)
        at Function.Module._load (module.js:470:3)
        at Function.Module.runMain (module.js:657:10)
        at startup (bootstrap_node.js:191:16)
        at bootstrap_node.js:613:3
  ...
```

`new ModuleJob` needs a `moduleProvider` as the third argument.

Reply:
I think you can just use a function that returns a never-resolving `Promise` for this